pkg/pgmodel/ingestor/parser.go

```diff
@@ -50,9 +62,20 @@ func (d *Parser) ParseData(tts []prompb.TimeSeries) (map[string][]model.Samples,
 			continue
 		}
 
+		var (
+			err    error
+			labels = t.Labels
+		)
+		if d.wAuth != nil {
+			labels, err = d.wAuth.VerifyAndApplyTenantLabel(tenant, t.Labels)
```

Comment:
I don't think we should be verifying the tenant here. Its too far down the ingestion pipeline, it should be verified at the API level and after validation, pushed down to the parser to apply the necessary tenant label.

Reply:
Verification was first at `write.go` but since we want to support tenants via external labels as well, hence things have to get verified here.